src/rust/engine/process_execution/src/remote.rs

```diff
@@ -844,15 +846,80 @@ fn maybe_add_workunit(
   }
 }
 
-pub fn make_execute_request(
+fn make_wrapper_for_append_only_caches(
+  caches: &BTreeMap<CacheName, RelativePath>,
+  base_path: &str,
+) -> Result<String, String> {
+  let mut script = String::new();
+  writeln!(&mut script, "#!/bin/sh").map_err(|err| format!("write! failed: {err:?}"))?;
+  for (cache_name, path) in caches {
+    writeln!(
+      &mut script,
+      "mkdir -p '{}/{}'",
+      base_path,
+      cache_name.name()
+    )
+    .map_err(|err| format!("write! failed: {err:?}"))?;
+    writeln!(
+      &mut script,
+      "ln -s '{}/{}' '{}'",
+      base_path,
+      cache_name.name(),
+      path.as_path().to_string_lossy()
+    )
+    .map_err(|err| format!("write! failed: {err}"))?;
+  }
+  writeln!(&mut script, "exec \"$@\"").map_err(|err| format!("write! failed: {err:?}"))?;
+  Ok(script)
+}
+
+pub async fn make_execute_request(
   req: &Process,
   instance_name: Option<String>,
   cache_key_gen_version: Option<String>,
+  store: &Store,
 ) -> Result<(remexec::Action, remexec::Command, remexec::ExecuteRequest), String> {
+  const CACHES_WRAPPER: &str = "./caches_wrapper";
+
+  // Implement append-only caches by running a wrapper script before the actual
+  let wrapper_script_digest_opt = if req.append_only_caches.is_empty() {
+    None
+  } else {
+    if req.working_directory.is_some() {
+      return Err(
+        "TODO-DEV: Append-only caches and working directory cannot be set together in remote execution.".into()
+      );
+    }
+    // TODO-WIP: Make the base path configurable.
```

Comment:
I'd recommend a global option, not the remote_environment target. That's because we use the same remote server for all remote_environment targets, and it seems desirable for them to share the same named cache. Also simpler modeling for users imo.

Reply:
This behavior also needs to be disableable, since not all remote execution environments will support it.

FWIW: I still think that I prefer explicit support in the backend rather than doing this in the client.